mono/utils/mono-utils-debug.c

```diff
@@ -0,0 +1,111 @@
+/* mono-utils-debug.c
+ *
+ * Copyright 2018 Microsoft
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+*/
+
+#include <config.h>
+#include <glib.h>
+#include "mono-utils-debug.h"
+
+#if defined (_WIN32)
+
+#include <windows.h>
+
+gboolean
+mono_is_usermode_native_debugger_present (void)
+{
+	// This is just a few instructions and no syscall. It is very fast.
+	// Kernel debugger is detected otherwise and is also useful for usermode debugging.
+	// Mono managed debugger is detected otherwise.
+	return IsDebuggerPresent () != FALSE;
+}
+
+#else
+
+#include <unistd.h>
+#include <errno.h>
+#include <fcntl.h>
+#if defined (__APPLE__)
+#include <sys/sysctl.h>
+#endif
+#if defined (__NetBSD__)
+#include <kvm.h>
+#endif
+
+static gboolean
+mono_is_usermode_native_debugger_present_slow (void)
+// PAL_IsDebuggerPresent https://github.com/dotnet/coreclr/blob/master/src/pal/src/init/pal.cpp
```

Comment:
Please don't link to the master branch as it'll get broken when code is moved around.

You can press the `y` key on the page (https://help.github.com/articles/getting-permanent-links-to-files/) to get a permalink for a specific commit of the file.

Reply:
Good point, sorry, thank you. Let's see if change is approved otherwise?